aws-rds-dbproxytargetgroup/src/main/java/software/amazon/rds/dbproxytargetgroup/CreateHandler.java

```diff
@@ -94,6 +94,12 @@
                     if (!callbackContext.isAllTargetsHealthy()) {
                         boolean allTargetsHealthy = checkTargetHealth(model);
 
+                        try {
+                            Thread.sleep(Constants.POLL_RETRY_DELAY_IN_MS);
+                        } catch (InterruptedException e) {
+                            throw new RuntimeException(e);
```

Comment:
If we're catching an `InterruptedException`, we should set the `interrupt` flag to true and then throw.

```
 ... {
  Thread.currentThread().interrupt();
  throw new RuntimeException(e);
}
```

Reply:
Pattern is from CFN public documentation for a handler